library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/ApplicationServerSetsRelBrandDoctrineRepository.php

```diff
@@ -43,4 +44,42 @@ public function findByBrandId(int $brandId): array
 
         return $response;
     }
+
+    /**
+     * @return int[]
+     */
+    public function getApplicationServerSetIdsByBrandAdmin(AdministratorInterface $admin): array
+    {
+        if (!$admin->isBrandAdmin()) {
+            throw new \DomainException('User must be brand admin');
+        }
+
+        $qb = $this->createQueryBuilder('self');
+        $expression = $qb->expr();
+
+        $brandId = $admin->getBrand()?->getId();
+        if (is_null($brandId)) {
+            return [];
+        }
+
+        $qb
+            ->select(
+                'IDENTITY(self.applicationServerSet) as applicationServerSet'
+            )
+            ->where(
+                $expression->eq(
+                    'self.brand',
+                    $brandId,
+                )
+            );
+
+        $results = $qb->getQuery()->getArrayResult();
+
+        $response = [];
+        foreach ($results as $result) {
+            $response[] = (int) $result['applicationServerSet'];
+        }
+
+        return $response;
```

Comment:
Do you agree that this logic should be in the service implementation and not the repository method? Besides, I don't realize why this casting is needed here

Reply:
This is called from API, so I don't have a chance to work on the caller side.